### PR TITLE
Fix graph command when plugins are involved

### PR DESCRIPTION
**Pull request by bolismauro**
Resolves https://github.com/tuist/tuist/issues/2506

### Short description 📝

The PR fixes https://github.com/tuist/tuist/issues/2506
The applied fix consists in loading the plugins following what [Generator.loadProject](https://github.com/tuist/tuist/blob/main/Sources/TuistKit/Generator/Generator.swift#L100) does.

Given that this is my very first PR, please let me know if there is a better way to do this. I tried to mimic what a class that was at the same level of the architecture was doing. However, I'm not sure it is the right approach.

An additional comment is related to the changes of the `GraphService` init. The `manifest loader` should be shared among all pieces, and I didn't find a better way to do it, if not splitting the two `init` (the live and the testing ones). Please let me know if there is a better way.

Finally, I tried to figure out how to properly test it, but I think I would need a little bit of help here. I tried to look at `GraphServiceTests`, but it hasn't been that helpful unfortunately.

#### Acceptance tests
I tried to run acceptance tests locally, but they fail. I tried to switch back to `main` and they fail there too. I think there is something wrong with my ruby configuration

Screen Shot 2021-02-12 at 19 44 10

### Checklist ✅

- [X] The code architecture and patterns are consistent with the rest of the codebase.
- [X] The changes have been tested following the [documented guidelines](https://tuist.io/docs/contribution/testing-strategy/).
- [x] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [X] In case the PR introduces changes that affect users, the documentation has been updated.

### CODE REVIEW

**Review by fortmarek — COMMENTED** [review]
Good catch @bolismauro! Could you maybe add a plugin to `ios_workspace_with_microfeature_architecture` fixture, so we check this in an acceptance test?

**Review by luispadron — APPROVED** [review]
Nice, completely overlooked this one! We should add a fixture to test against this breaking in the future.

**Comment by bolismauro** [comment]
@fortmarek sure. I'll do it.
Do you have any idea why acceptance tests are failing on my mac? It would be great being able to have them to make sure test doesn't pass without the fix and passes with it

**Comment by bolismauro** [comment]
Plugin added to the fixture. I also discovered by acceptance tests don't work on my computer (see #2502 )

**Comment by bolismauro** [comment]
The failed test seems to be completely unrelated
```
2021-02-14T13:16:54.2446850Z Test Case '-[TuistAsyncQueueTests.AsyncQueueTests test_start_sentPersistedEventIsThenDeleted]' started.
2021-02-14T13:16:58.1567920Z /Users/runner/work/tuist/tuist/Tests/TuistAsyncQueueTests/AsyncQueueTests.swift:291: error: -[TuistAsyncQueueTests.AsyncQueueTests test_start_sentPersistedEventIsThenDeleted] : failed - Sent persisted event was then not deleted
2021-02-14T13:16:58.1580360Z Test Case '-[TuistAsyncQueueTests.AsyncQueueTests test_start_sentPersistedEventIsThenDeleted]' failed (3.913 seconds).
2021-02-14T13:16:58.1582160Z Test Suite 'AsyncQueueTests' failed at 2021-02-14 13:16:58.158.
2021-02-14T13:16:58.1582830Z 	 Executed 10 tests, with 1 failure (0 unexpected) in 3.932 (3.933) seconds
```

**Comment by fortmarek** [comment]
> The failed test seems to be completely unrelated

It does seem to be unrelated, indeed, I tried to rerun the job, it is possible the test is flaky. If it is, it's something we should investigate and fix.

**Comment by bolismauro** [comment]
Let me know if there is something else to do for this PR!

**Comment by luispadron** [comment]
@bolismauro Looks good to merge once we resolve conflicts.

**Comment by bolismauro** [comment]
@luispadron done! :yey:

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,9 +8,13 @@ Please, check out guidelines: https://keepachangelog.com/en/1.0.0/
...
 - Add ODR support [#2490](https://github.com/tuist/tuist/pull/2490) by [@DimaMishchenko](https://github.com/DimaMishchenko)
 
+### Fixed
+
+- Fix crash when using `tuist graph` in a project that leverages plugins [#2507](https://github.com/tuist/tuist/pull/2507) by [@bolismauro](https://github.com/bolismauro).
+
 ### Changed
 
-- Replace `ExpressibleByStringLiteral` with `ExpressibleByStringInterpolation` for `ProjectDescription` objects by [@DimaMishchenko](https://github.com/DimaMishchenko)
+- Replace `ExpressibleByStringLiteral` with `ExpressibleByStringInterpolation` for `ProjectDescription` objects by [@DimaMishchenko](https://github.com/DimaMishchenko
```
Comment (fortmarek) [inline CHANGELOG.md:17]: the deleted bracket is probably a mistake

**Review by fortmarek — APPROVED** [review]

**Review by pepibumur — APPROVED** [review]

**Pull request merged**